Update documentation (#194)
GIScience/ohsome-api: docs/http-response-status.rst

HTTP Response Status
====================

List of HTTP status codes and possible messages.

2xx success
-----------

``200 OK`` - standard response for successful GET or POST requests.

.. note:: If an error occurs during a data-extraction request, the result will be a broken GeoJSON containing the error JSON object at the end. In this case, the ohsome API will give back a 200 response status.

4xx client errors
-----------------

``400 Bad Request`` - the ohsome API cannot process the request due to a client error. This can occur because a parameter is not given in the correct format, mandatory parameters are not present, unknown or not valid parameters are present, parameters are not unique, malformed GeoJSON, unsopported content-type reader, wrong parameter values or other malformed requests.

``404 Not Found``-  the requested resource could not be found because the given boundary or time are not within the underlying osh-data.

> ``413 Payload Too Large`` - the request is too large in respect to the given timeout parameter.

Reviewer: imho this could be worded a bit simpler/better? Specifically _"too large"_ means in our case that it _"took too long"_, and not all queries have a _"given timeout"_, but use the default one:

```suggestion
``413 Payload Too Large`` - the request took too long to compute.
```

Author: OK, but it seems to me it doesn't inform the user in respect of what it took too long. What do you think about `the request took too long to compute in respect of the given or default timeout.`?

Reviewer: that should work

* "There cannot be more input values in the values|values2 than in the keys|keys2 parameter, as values_n must fit to keys_n."
* "Parameter 'types' (and 'types2') cannot have more than 4 entries."
* "The given timeout does not fit to its format. Please give one value in seconds and use a point as the decimal delimiter, if needed."
* "The given timeout is too long. It has to be shorter than [timeout] seconds"
* "You need to give at least two timestamps or a time interval for this resource."
* "The given parameter [parameter] can only contain the values 'true', 'yes', 'false', or 'no'."
* "Your provided boundary parameter (bboxes, bcircles, or bpolys) does not fit its format, or you defined more than one boundary parameter."
* "The geometry of your given boundary input could not be parsed for the creation of the response GeoJSON."
* "The keys, values and types parameters must be empty, when you set the filter parameter."
* "The properties parameter of this resource can only contain the values 'tags' and/or 'metadata' and/or 'contributionTypes' and/or 'unclipped'."
* "The properties parameter of this resource can only contain the values 'tags' and/or 'metadata' and/or 'unclipped'."

> * "Parameter 'types' (and 'types2') can only have 'node' and/or 'way' and/or 'relation' OR 'point' and/or 'line' and/or 'polygon' and/or 'other'"

Reviewer: I don't know how useful a list of possible messages is, to be honest. Since they are just listed here plainly without further explanation, there's not really much users could learn from this list, could they? If anything, the list would be quite cumbersome to maintain (i.e. keep up-to-date) for us.

Instead one could simply write that: _“The error message contains further details about why the query failed.”

Or what was your reasoning behind including it? Maybe I'm simply not seeing it at the moment.

Author: I understood #150 in that way. @joker234 Am I wrong? If yes, no problem, I can change it summarizing the reasons for every  HTTP response code of the ohsome API.

Reviewer: hm, ok. I would say it might be better to mention the possible cases in a slightly abstracted way, instead of a plain verbatim list. e.g. something like this:

```
* ``400 Bad Request`` - the ohsome API cannot or will not process the request due to a client error. This can occur because a parameter is not given in the correct format,  mandatory parameters are not present or other malformed requests. The error message contains further details about why the query failed.
* ``404 Not Found``-  the requested resource could not be found. This can occur when an invalid resource is requested, or when the provided boundary or time parameter does is not covered by the underlying data OSHDB database.
* ``413 Payload Too Large`` - the request is larger than the ohsome API is willing or able to process. Occurs when the given query took takes too long to compute.
```

Reviewer 2: I'm not quite sure what the intention should be. I would agree it doesn't make sense to joust copy & paste all error messages as you have to keep track of them, when you change the ohsome API's code. But maybe a little bit better explanation in the ohsome API documentation would be nice. Alternatively one could think about having more actual error codes in one list as the ORS has. https://github.com/GIScience/openrouteservice/blob/master/openrouteservice/docs/services/error_codes.md

Author: I have updated this part trying to improve the explanation without the actual messages.
